bluetooth/src/iosMain/kotlin/scanner/Scanner.kt

```diff
@@ -60,53 +69,87 @@ actual class DefaultScanner internal constructor(
             scanner.handleDeviceConnected(didConnectPeripheral.identifier)
         }
 
-        override fun centralManager(central: CBCentralManager, didDisconnectPeripheral: CBPeripheral, error: NSError?) {
+        override fun centralManager(
+            central: CBCentralManager,
+            didDisconnectPeripheral: CBPeripheral,
+            error: NSError?
+        ) {
             scanner.handleDeviceDisconnected(didDisconnectPeripheral.identifier)
         }
 
-        override fun centralManager(central: CBCentralManager, didFailToConnectPeripheral: CBPeripheral, error: NSError?) {
+        override fun centralManager(
+            central: CBCentralManager,
+            didFailToConnectPeripheral: CBPeripheral,
+            error: NSError?
+        ) {
             scanner.handleDeviceDisconnected(didFailToConnectPeripheral.identifier)
         }
     }
 
     private val enabledQueue = dispatch_queue_create("ScannerMonitorEnabled", null)
     private val scanQueue = dispatch_queue_create("ScannerScanning", null)
-    private val pairedDevicesQueue = dispatch_queue_create("ScannerPairedDevices", null)
     override val isSupported: Boolean = true
-    private val centralManagers = sharedMutableListOf<CBCentralManager>()
-    private val discoveringDelegates = sharedMutableListOf<CBCentralManagerDelegateProtocol>()
-    private val activeDelegates = sharedMutableSetOf<CBCentralManagerDelegateProtocol>()
-    override val bluetoothEnabledMonitor: BluetoothMonitor = BluetoothMonitor.Builder { CBCentralManager(null, enabledQueue, emptyMap<Any?, Any>()) }.create()
+    override val bluetoothEnabledMonitor: BluetoothMonitor = BluetoothMonitor.Builder {
+        CBCentralManager(null, enabledQueue, emptyMap<Any?, Any>())
+    }.create()
+
+    private val centralManager = AtomicReference<CBCentralManager?>(null)
+    private val centralManagerDelegate = AtomicReference<CBCentralManagerDelegateProtocol?>(null)
+    private val centralManagerMutex = Mutex()
+    private suspend fun getCentralManager(): CBCentralManager = centralManagerMutex.withLock {
+        if (centralManager.value == null) {
+            createCentralManager()
+        } else {
+            debug(TAG) { "Access CBCentralManager" }
+            centralManager.value!!
+        }
+    }
 
-    private suspend fun scan(filter: UUID? = null) {
-        val centralManager = CBCentralManager(null, scanQueue)
-        centralManagers.add(centralManager)
+    private suspend fun createCentralManager(): CBCentralManager {
+        debug(TAG) { "Create CBCentralManager" }
+        val manager = CBCentralManager(null, scanQueue)
         val awaitPoweredOn = EmptyCompletableDeferred()
-        val delegate = PoweredOnCBCentralManagerDelegate(this, awaitPoweredOn)
-        discoveringDelegates.add(delegate)
-        centralManager.delegate = delegate
+        val delegate = PoweredOnCBCentralManagerDelegate(this, awaitPoweredOn).freeze()
```

Comment:
Question out of curiosity and probably to @Daeda88: 
What difference does it make that this code first creates `CBCentralManager` then `PoweredOnCBCentralManagerDelegate` and then assigns delegate to manager?

For me it would seem more natural to first create `PoweredOnCBCentralManagerDelegate` and then `CBCentralManager` with a delegate in init.

Reply:
@jtomanik fixed it and it doesn't break anything. Will test a bit more however